services/supabase/migrations/20260724100000_create_referral_profiles.sql

```
@@ -0,0 +1,23 @@
+-- Referral profiles: stores activated referrer codes for lookup and sharing
+CREATE TABLE referral_profiles (
+  user_id UUID PRIMARY KEY REFERENCES auth.users(id) ON DELETE CASCADE,
+  referral_code TEXT NOT NULL UNIQUE,
+  stellar_address TEXT,
+  activated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  on_chain_ready BOOLEAN NOT NULL DEFAULT false,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE INDEX idx_referral_profiles_referral_code ON referral_profiles(referral_code);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)